File: 백준/Silver/5635. 생일/생일.py
T = int(input())
list1 = []
for i in range(T):
    name, day, month, year = input().split()
    day, month, year = map(int, (day, month, year))
    list1.append((year, month, day, name))
list1.sort()
print(list1[-1][3])
print(list1[0][3])

File: 백준/Silver/5635. 생일/생일.cc
#include <iostream>
#include <algorithm>

using namespace std;

int N;

struct Person{
    string name;
    int day, month, year;

    bool operator<(const Person& other) const {
        if (year != other.year)
            return year < other.year;
        if (month != other.month)
            return month < other.month;
        return day < other.day;
    }
};

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    
    cin >> N;

    Person arr[101];
    for(int i = 0; i < N; i++){
        cin >> arr[i].name >> arr[i].day >> arr[i].month >> arr[i].year;
    }

    sort(arr, arr + N);
    
    cout << arr[N - 1].name << '\n';
    cout << arr[0].name;
}